app/routes/polls/components/PollEditor.js

```diff
@@ -0,0 +1,195 @@
+      {fields.map((option, i) => (
+        <li className={styles.optionField} key={i}>
+          <Field
+            name={`${option}.name`}
+            label={`Valg nr. ${i + 1}`}
+            placeholder={`Valg ${i + 1}`}
+            component={TextInput.Field}
+            required
+          />
+          <ConfirmModalWithParent
+            title="Slett valg"
+            message="Er du sikker på at du vil slette dette valget?"
+            onConfirm={() => Promise.resolve(fields.remove(i))}
+            closeOnConfirm
+            className={styles.deleteOption}
+          >
+            <Icon name="trash" className={styles.deleteOption} />
+          </ConfirmModalWithParent>
+        </li>
+      ))}
+    </ul>
+
+    <Link
+      key="addNew"
+      onClick={() => {
+        fields.push({});
+      }}
+      className={styles.addOption}
+    >
+      <Icon name="add-circle" size={25} /> Valg
+    </Link>
+  </div>
+);
+
+class EditPollForm extends Component<Props, *> {
+  render() {
+    const {
+      pristine,
+      submitting,
+      handleSubmit,
+      editing,
+      deletePoll
+    } = this.props;
+
+    return (
+      <Content>
+        {!editing && (
+          <NavigationTab title="Ny avstemning">
+            <NavigationLink to="/polls">Tilbake</NavigationLink>
+          </NavigationTab>
+        )}
+        <Form onSubmit={handleSubmit}>
+          <Field
+            name="title"
+            label="Spørsmål"
+            placeholder="Hva er din favorittrett?"
+            component={TextInput.Field}
+            required
+          />
+          <span />
+          <Field
+            name="description"
+            label="Beskrivelse"
+            placeholder="Mer info..."
+            component={TextArea.Field}
+          />
+          <Field
+            name="pinned"
+            label="Vis på forsiden"
+            component={CheckBox.Field}
+          />
+          <Field
+            name="tags"
+            label="Tags"
+            filter={['tags.tag']}
+            placeholder="Skriv inn tags"
+            component={SelectInput.AutocompleteField}
+            multi
+            tags
+            shouldKeyDownEventCreateNewOption={({
+              keyCode
+            }: {
+              keyCode: number
+            }) => keyCode === 32 || keyCode === 13}
+          />
+          <FieldArray
+            name="options"
+            component={renderOptions}
+            rerenderOnEveryChange={true}
+          />
+          <Button
+            disabled={pristine || submitting}
+            submit
+            className={styles.submitButton}
+          >
+            {editing ? 'Endre avstemning' : 'Lag ny avstemning'}
+          </Button>
+          {editing && (
+            <ConfirmModalWithParent
+              title="Slett avstemning"
+              message="Er du sikker på at du vil slette avstemningen?"
+              onConfirm={() => Promise.resolve(deletePoll())}
```

Comment:
`deletePoll` is an action and therefore it already returns a promise. Therefore, instead of writing this line, you can replace it with `onConfirm={deletePoll}`.

The actions in lego returns promises because we use `redux-thunk`. 😄